Assets/Scripts/Runtime/Character/Hero.cs

```diff
@@ -321,4 +317,22 @@ private void disableThirdPersonController()
             starterAssetsInputs.DisableInputs();
         }
     }
+
+    public struct NavMeshUtility
+    {
+        private const int MAX_NAVMESH_DISTANCE = 50;
+        private static NavMeshHit navMeshHit;
```

Comment:
why static?

if static remove this field and consider doing a static class and function

but maybe just put this hit info in the public field and have an instance...